spoj_practice/spoj2/spoj2.py

```diff
@@ -7,56 +6,26 @@
 # N < 1,000,000,000
 # Find all primes between a given n, m
 
-
-# handle stdin/stdout and unit testing
-
-
-
 # returns a list of all the primes in between
+def is_divisible (i, some_prime):
+    return i % some_prime == 0
+
 def find_primes_between (n, m):
-    # divide by all odd numbers, counter it up
-    # divide by all primes before it
-    # divide by all other primes between n, m
-    # make sure we start with an odd number
-    
-    # primes to check against
-    # 1 is not a prime number, the loop excludes even numbers thus 2 does not need to be added
-    primes = [3]
-    primes_between = []
-    
-    # unique cases
-    if m < 2:
-        primes_between.append(2)
-    if m <= 3:
-        primes_between.append(3)
-    
+    primes = [2]
     # check all primes
-    for i in xrange(3, n+1, 2):
-        # check if number is prime
-        for prime in primes:
-            # if is not prime, move onto next number i
-            if i%prime is 0:
-                break
-        # if the number is in fact prime,
-        else:
+    for i in xrange(2, n+1):
+        # in brackets, evaluates is_divisible calculation for a given i against all known primes (denoted by for prime in primes). think about it like the the for loop is being evaluated with the is_divisible function nested within. all of the results are compiled into a list of booleans. this list of boolean values (again, for a given i) is returned to the function 'any', which then checks for if any of those are true. if so, is_composite is set to true.
+        is_composite = any(is_divisible(i, prime) for prime in primes)
+        # then we say when is_composite is false for a given i, aka its a prime, then we append it to the list of primes
+        if not is_composite:
             primes.append(i)
-            if i >= m:
-                primes_between.append(i)
-    
-    return primes_between 
+    return primes
 
 num_of_lines = int(stdin.readline())
 for i in xrange(num_of_lines):
-    # numbers = stdin.readline().split(" ")
-    # m = numbers[0]
-    # n = numbers[1]
-    # instead, do:
-    m, n = stdin.readline().split(' ')
-    m = int(m)
-    n = int(n)
-    prime = []
+    m, n = map(int, stdin.readline().split(' '))
     primes = find_primes_between(n, m)
```

Comment:
Does the list of prime changes for each test case? Do we need to compute this every time, or can we even save the primes from a previous test case for the next case?